src/engine/Service.cpp

```diff
@@ -455,3 +454,34 @@ void Service::throwErrorWithContext(std::string_view msg,
       (last100.empty() ? "'"
                        : absl::StrCat(", last 100 bytes: '", last100, "'"))));
 }
+
+// ____________________________________________________________________________
+std::optional<std::string> Service::idToValueForValuesClause(
+    const Index& index, Id id, const LocalVocab& localVocab) {
+  using enum Datatype;
+  const auto& optionalStringAndXsdType =
+      ExportQueryExecutionTrees::idToStringAndType(index, id, localVocab);
+  if (!optionalStringAndXsdType.has_value()) {
+    return std::nullopt;
+  }
```

Comment:
```suggestion
  AD_CORRECTNESS_CHECK(optStringAdnType.has_value());
```
(We should refactor this function to not even return an `optional` in the first place, but that is for another PR).

Reply:
Sorry, i was mislead.  There is one case missing, if the id is undefined, 
then you have to return the string `"UNDEF"` and put it into the values clause. In particular this behavior is different from that of the blank nodes (see my comment below).

Reply:
So something like ` if not has_value, then assert that the id is undef and return "UNDEF"`